// react-customizable-carousel/lib/Dots/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Dots = void 0;

var _react = _interopRequireWildcard(require("react"));

var _DotsDot = require("./DotsDot");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

class Dots extends _react.PureComponent {
  get cssClass() {
    const className = this.props.className;
    return `rcc-Dots ${className || ''}`;
  }

  render() {
    const _this$props = this.props,
          onDotClick = _this$props.onDotClick,
          currentIndex = _this$props.currentIndex,
          itemsCount = _this$props.itemsCount,
          dotComponent = _this$props.dotComponent,
          dotWrapperClassName = _this$props.dotWrapperClassName;
    const indexes = [...Array(itemsCount).keys()];
    return _react.default.createElement("div", {
      className: this.cssClass
    }, indexes.map(index => _react.default.createElement(_DotsDot.DotsDot, {
      key: index,
      index: index,
      component: dotComponent,
      onClick: onDotClick,
      current: currentIndex === index,
      className: dotWrapperClassName
    })));
  }

}

exports.Dots = Dots;
