// react-customizable-carousel/lib/List/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.List = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactMeasure = require("react-measure");

var _ListItems = require("./ListItems");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

class List extends _react.PureComponent {
  get cssClass() {
    const className = this.props.className;
    return `rcc-List ${className || ''}`;
  }

  render() {
    const _this$props = this.props,
          items = _this$props.items,
          measureRef = _this$props.measureRef,
          showItemsCount = _this$props.showItemsCount,
          width = _this$props.contentRect.bounds.width,
          currentIndex = _this$props.currentIndex,
          direction = _this$props.direction,
          transitionDuration = _this$props.transitionDuration,
          translateX = _this$props.translateX,
          scrollItems = _this$props.scrollItems,
          onNext = _this$props.onNext,
          onPrev = _this$props.onPrev,
          enableDragScroll = _this$props.enableDragScroll,
          enable3d = _this$props.enable3d,
          nextIndex = _this$props.nextIndex,
          inTransition = _this$props.inTransition,
          effectOf3d = _this$props.effectOf3d;
    return _react.default.createElement("div", {
      className: this.cssClass,
      ref: measureRef
    }, _react.default.createElement(_ListItems.ListItems, {
      itemWidth: width / showItemsCount,
      showItemsCount: showItemsCount,
      currentIndex: currentIndex,
      nextIndex: nextIndex,
      direction: direction,
      transitionDuration: transitionDuration,
      translateX: translateX,
      scrollItems: scrollItems,
      onNext: onNext,
      onPrev: onPrev,
      enableDragScroll: enableDragScroll,
      enable3d: enable3d,
      effectOf3d: effectOf3d,
      inTransition: inTransition
    }, items));
  }

}

exports.List = List;

var _default = (0, _reactMeasure.withContentRect)('bounds')(List);

exports.default = _default;
