// react-customizable-carousel/es/Dots/DotsDot/index.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { PureComponent } from 'react';
export class DotsDot extends PureComponent {
  constructor() {
    super(...arguments);

    _defineProperty(this, "handleClick", () => {
      const _this$props = this.props,
            onClick = _this$props.onClick,
            index = _this$props.index;
      onClick(index);
    });
  }

  get cssClass() {
    const className = this.props.className;
    return `rcc-Dot ${className || ''}`;
  }

  render() {
    const _this$props2 = this.props,
          current = _this$props2.current,
          component = _this$props2.component;
    const CustomContent = component;
    return React.createElement("span", {
      className: this.cssClass,
      onClick: this.handleClick
    }, component ? React.createElement(CustomContent, {
      current: current
    }) : React.createElement(DefaultContent, {
      current: current
    }));
  }

}

const DefaultContent = (_ref) => {
  let current = _ref.current;
  return current ? React.createElement("div", {
    className: "rcc-Circle-icon rcc-Circle-icon--filled"
  }) : React.createElement("div", {
    className: "rcc-Circle-icon"
  });
};
