// react-customizable-carousel/lib/Arrow/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Arrow = exports.ARROW_TYPES = void 0;

var _react = _interopRequireWildcard(require("react"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

const ARROW_TYPES = {
  prev: 'prev',
  next: 'next'
};
exports.ARROW_TYPES = ARROW_TYPES;

class Arrow extends _react.PureComponent {
  get cssClass() {
    const _this$props = this.props,
          className = _this$props.className,
          arrowType = _this$props.arrowType;
    const customClassName = !className ? '' : `${className} ${arrowType ? `${className}--${arrowType}` : ''}`;
    return `rcc-Arrow ${arrowType ? `rcc-Arrow--${arrowType}` : ''} ${customClassName}`;
  }

  render() {
    const _this$props2 = this.props,
          onClick = _this$props2.onClick,
          component = _this$props2.component,
          arrowType = _this$props2.arrowType;
    const ArrowComponent = component;
    return _react.default.createElement("div", {
      className: this.cssClass
    }, _react.default.createElement(ArrowComponent, {
      onClick: onClick,
      arrowType: arrowType
    }));
  }

}

exports.Arrow = Arrow;
